auto3dseg/algorithm_templates/segresnet/scripts/segmenter.py

```diff
@@ -464,47 +443,51 @@ def __init__(
         elif torch.cuda.is_available():
             torch.backends.cudnn.benchmark = True
 
-        if config.get("anisotropic_scales", False) and 'SegResNetDS' in config["network"]["_target_"]:
-            parser.config["network"]["resolution"] = config["resample_resolution"]
-            parser.parse(reset=True)
-            print('Using anisotripic scales', parser['network'])
-
-        model = parser.get_parsed_content("network")
-
-        if config["pretrained_ckpt_name"] is not None:
-            self.checkpoint_load(ckpt=config["pretrained_ckpt_name"], model=model)
 
-        model = model.to(self.device)
+        self.model = self.setup_model()
 
-        if self.distributed:
-            model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
-            model = DistributedDataParallel(
-                module=model, device_ids=[rank], output_device=rank, find_unused_parameters=False
-            )
-
-        if rank == 0:
-            pytorch_total_params = sum(p.numel() for p in model.parameters() if p.requires_grad)
-            print("Total parameters count", pytorch_total_params, "distributed", self.distributed)
-
-        self.model = model
         self.loss_function = parser.get_parsed_content("loss")
         self.loss_function = DeepSupervisionLoss(self.loss_function)
-
         self.acc_function = DiceHelper(sigmoid=config["sigmoid"])
         self.grad_scaler = GradScaler(enabled=config["amp"])
 
+
         if parser.get("sliding_inferrer") is not None:
             self.sliding_inferrer = parser.get_parsed_content("sliding_inferrer")
         else:
-            self.sliding_inferrer = SlidingWindowInferer(
-                roi_size=config["roi_size"],
-                sw_batch_size=1,
-                overlap=0.625,
-                mode="gaussian",
-                cache_roi_weight_map=True,
-                progress=False,
-                cpu_thresh=512**3 // config["output_classes"],
-            )
+
+            try:
+                if __package__ in (None, ""):
+                    from am_sliding_inferrer import AMSlidingWindowInferer
```

Comment:
Do we still need the `am_sliding_inferer` because `SlidingWindowInferer` is suboptimal?

Reply:
removed, there is a new class there SlidingWindowInfererAdap (not merged to core yet), so you'll see try/exept for now (waiting for PR to be merged)